homeassistant/components/tplink/entity.py

```diff
@@ -181,24 +173,10 @@ def _get_unique_id(self) -> str:
             return unique_id
 
         # For legacy sensors, we construct our IDs from the entity description
-        if self.entity_description is not None:
+        if isinstance(self, SensorEntity) and self.entity_description is not None:
```

Comment:
the base entity should never know about the details of a platform

Reply:
Done in https://github.com/home-assistant/core/pull/120025